pkgs/development/libraries/libvncserver/default.nix

```diff
@@ -32,11 +25,12 @@ stdenv.mkDerivation {
   preConfigure = ''
     sed -e 's@/usr/include/linux@${stdenv.cc.libc}/include/linux@g' -i configure
   '';
+  propagatedBuildInputs = [ zlib ];
```

Comment:
If you are making `zlib` propagated, then it is not needed in normal `buildInputs`. And if you are splitting `buildInputs` already, `autoreconfHook` and `pkgconfig` are native (i.e. executed in the build-time), nowadays after all the cross-compilation work these go into `nativeBuildInputs`.